tc/autotuner/options_cache-inl.h

```diff
@@ -317,9 +318,41 @@ void OptionsCache<Backend>::fromProtobuf(
   }
 }
 
-} // namespace autotune
+template <typename Backend>
+std::vector<typename Backend::MappingOptionsType> loadTopKFromCache(
+    const std::string& tc,
+    const std::string& entryPoint,
+    const std::string& cacheFileName,
+    const std::vector<const DLConstTensor*>& inputs,
+    const size_t numCandidates) {
+  LOG(INFO) << "Loading proto from: " << cacheFileName << std::endl;
+  if (cacheFileName.empty()) {
+    return {};
+  }
+  OptionsCache<Backend> optionsCache;
+  optionsCache.loadCacheFromFile(cacheFileName);
+  auto outputs = tc::inferOutputTensorInfo(tc, entryPoint, inputs);
+  return optionsCache.getTopKOptions(
+      lang::canonicalTc(tc::detail::parse(tc).at(entryPoint)),
+      tc::makeTensorInfoVector(inputs),
+      outputs,
+      Backend::backendString(),
+      numCandidates);
+}
 
-inline std::string makeOptionsFilename(const std::string& fn) {
-  return fn + ".options";
+template <typename Backend>
+void storeTopKToCache(
+    const std::shared_ptr<OptionsCache<Backend>>& cache,
+    const std::string& cacheFilename,
+    uint32_t count) {
+  if (cacheFilename.empty()) {
+    LOG(INFO) << "No filepath provided, not saving cache" << std::endl;
+  } else {
+    LOG(INFO) << "Dumping cache to " << cacheFilename << std::endl;
+    cache->pruneKeepTopK(count);
```

Comment:
Guessing from the function name, I'd expect it to store top-k options to the given file but keep the cache object itself intact.  Here it looks like it prunes _from the cache_.  Please address this before landing.